From 67c3aa3d256bda575911d3079507185f7d649287 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Wed, 28 Oct 2015 16:24:40 +0800
Subject: [PATCH] MLK-11791 pxp: reinitialize lut_state to 0 to fix LUT unset
 issue

because the registers are reset to default value in interrupt handler,
we need reinitialize lut_stat to 0 otherwise the LUT registers will
not be set due to the following logic judgement in pxp_set_lut():

	if ((pxp->lut_state == lut_op) &&
		!(use_cmap && pxp_conf->proc_data.lut_map_updated))
		return;

	lut setting here...

	pxp->lut_state == lut_op;

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 drivers/dma/pxp/pxp_dma_v2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/dma/pxp/pxp_dma_v2.c b/drivers/dma/pxp/pxp_dma_v2.c
index 5225bcf..48d751d3 100644
--- a/drivers/dma/pxp/pxp_dma_v2.c
+++ b/drivers/dma/pxp/pxp_dma_v2.c
@@ -1319,6 +1319,7 @@ static irqreturn_t pxp_irq(int irq, void *dev_id)
 
 	if (list_empty(&head)) {
 		pxp->pxp_ongoing = 0;
+		pxp->lut_state = 0;
 		spin_unlock_irqrestore(&pxp->lock, flags);
 		return IRQ_NONE;
 	}
@@ -1349,6 +1350,7 @@ static irqreturn_t pxp_irq(int irq, void *dev_id)
 
 	complete(&pxp->complete);
 	pxp->pxp_ongoing = 0;
+	pxp->lut_state = 0;
 	mod_timer(&pxp->clk_timer, jiffies + msecs_to_jiffies(timeout_in_ms));
 
 	spin_unlock_irqrestore(&pxp->lock, flags);
-- 
1.8.0

